File: atoci/Arrays & Hashing/cpp/containsDuplicate.cpp
#include<iostream>
#include<unordered_map>
#include<vector>

using namespace std; 

bool containsDuplicate(const vector<int>& x) {
    unordered_map<int, int> mymap;

    for (auto &val : x) {
        mymap[val] = 0; 
    }
    
    for (const auto& val : x) {
        mymap[val]++; 
    }

    for (const auto& pair : mymap) {
        if (pair.second > 1) {
            return true;
        }
    }
    
    return false;
}


int main() {
    cout << containsDuplicate({1,2,3,4,5}) << endl;
    cout << containsDuplicate({1,2,3,4,5,6,7,8,9,1}) << endl;
    return 0;
}

File: atoci/Arrays & Hashing/typescript/containsDuplicate.ts
function containsDuplicate(x : number[]): boolean {

    const mymap : {[key: number]: number} = {};

    for (let index = 0; index < x.length; index++) {
        mymap[x[index]] = 0; 
    }

    for (let index = 0; index < x.length; index++) {
        mymap[x[index]]++; 
    }

    for (const key in mymap) {
        if (mymap[key] > 1) {
            return true
        }
    }

    return false
}

console.log(containsDuplicate([1,2,3,4,5]))
console.log(containsDuplicate([1,2,3,4,5,6,7,8,9,1]))